x/stakeibc/keeper/msg_server_instant_redeem_stake_test.go

```diff
@@ -0,0 +1,192 @@
+package keeper_test
+
+import (
+	sdkmath "cosmossdk.io/math"
+	"fmt"
+	epochtypes "github.com/Stride-Labs/stride/v5/x/epochs/types"
+	recordtypes "github.com/Stride-Labs/stride/v5/x/records/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/stretchr/testify/suite"
+)
+
+type InstantRedeemStakeState struct {
+	epochNumber         uint64
+	depositRecordAmount sdkmath.Int
+	hostZone            stakeibctypes.HostZone
+}
+
+type InstantRedeemStakeTestCase struct {
+	user         Account
+	zoneAccount  Account
+	initialState InstantRedeemStakeState
+	validMsg     stakeibctypes.MsgInstantRedeemStake
+}
+
+func (s *KeeperTestSuite) SetupInstantRedeemStake() InstantRedeemStakeTestCase {
+	unbondAmount := sdkmath.NewInt(1_000_000)
+	depositAmount := sdkmath.NewInt(500_000)
+	user := Account{
+		acc:           s.TestAccs[0],
+		atomBalance:   sdk.NewInt64Coin(IbcAtom, 10_000_000),
+		stAtomBalance: sdk.NewInt64Coin(StAtom, 10_000_000),
+	}
+	s.FundAccount(user.acc, user.atomBalance)
+	s.FundAccount(user.acc, user.stAtomBalance)
+
+	zoneAddress := stakeibctypes.NewZoneAddress(HostChainId)
+
+	zoneAccount := Account{
+		acc:           zoneAddress,
+		atomBalance:   sdk.NewInt64Coin(IbcAtom, 10_000_000),
+		stAtomBalance: sdk.NewInt64Coin(StAtom, 10_000_000),
+	}
+	s.FundAccount(zoneAccount.acc, zoneAccount.atomBalance)
+	s.FundAccount(zoneAccount.acc, zoneAccount.stAtomBalance)
+
+	hostZone := stakeibctypes.HostZone{
+		ChainId:        HostChainId,
+		HostDenom:      Atom,
+		IbcDenom:       IbcAtom,
+		RedemptionRate: sdk.NewDec(1.0),
+		Address:        zoneAddress.String(),
+		StakedBal:      unbondAmount,
+	}
+
+	epochTracker := stakeibctypes.EpochTracker{
+		EpochIdentifier: epochtypes.STRIDE_EPOCH,
+		EpochNumber:     1,
+	}
+
+	oldDepositRecord := recordtypes.DepositRecord{
+		Id:                 0,
+		DepositEpochNumber: 0,
+		HostZoneId:         "GAIA",
+		Amount:             depositAmount,
+	}
+
+	initialDepositRecord := recordtypes.DepositRecord{
```

Comment:
Makes sense to test the case with multiple DRs, but only one should have Status TRANSFER_QUEUE per host zone I believe

Reply:
Previous failed transfers when they time out will be set back to TRANSFER_QUEUE, so we will try again in the future.  So the case where we have multiple entries is when they are all this.

Reply:
Great point. Resolving